DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -0,0 +1,447 @@
+""" MoveReplica operation handler
+
+    This handler moves replicas from source SEs to target SEs. Replicas are first replicated to target SEs and then removed from the source SEs
+"""
+
+__RCSID__ = "$Id $"
+
+# # imports
+import re, os, string
+# # from DIRAC
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.FrameworkSystem.Client.MonitoringClient import gMonitor
+from DIRAC.Core.Utilities.Adler import compareAdler
+
+from DIRAC.DataManagementSystem.Client.DataManager                                import DataManager
+from DIRAC.DataManagementSystem.Agent.RequestOperations.DMSRequestOperationsBase  import DMSRequestOperationsBase
+
+from DIRAC.Resources.Storage.StorageElement                                       import StorageElement
+from DIRAC.Resources.Catalog.FileCatalog                                          import FileCatalog
+
+def filterReplicas( opFile, logger = None, dataManager = None ):
+  """ filter out banned/invalid source SEs """
+
+  if logger is None:
+    logger = gLogger
+  if dataManager is None:
+    dataManager = DataManager()
+
+  log = logger.getSubLogger( "filterReplicas" )
+  ret = { "Valid" : [], "NoMetadata" : [], "Bad" : [], 'NoReplicas':[], 'NoPFN':[] }
+  replicas = dataManager.getActiveReplicas( opFile.LFN )
+
+  if not replicas["OK"]:
+    log.error( 'Failed to get active replicas', replicas["Message"] )
+    return replicas
+  reNotExists = re.compile( r".*such file.*" )
```

Comment:
The answer to the question is yes ! I wrote cmpError specifically so that it works for old style and new style. You can give it a string. See the code https://github.com/DIRACGrid/DIRAC/blob/integration/Core/Utilities/DErrno.py#L239